util/interp.h

```diff
@@ -3,22 +3,139 @@
 // Copyright (c) 2018-present The Alive2 Authors.
 // Distributed under the MIT license that can be found in the LICENSE file.
 
+#include <iostream>
 #include <map>
 #include <memory>
 #include <string>
+#include <variant>
 #include <vector>
 
 namespace IR {
 class BasicBlock;
 class Function;
 class Input;
 class Value;
-}
+} // namespace IR
 
 namespace util {
 
 class ConcreteVal;
 
+struct ConcreteByte {
+  bool is_pointer;
+  // for now, for non pointer bytes, treat the entire byte as an i8 for poison
+  // purposes
+  std::variant<ConcreteValPointer, ConcreteValInt> value;
```

Comment:
It seems really wasteful to use `ConcreteValInt` here, since the value is always 8 bits. You just need an `uint8_t` for the value and another `uint8_t` for the poison bits.